core/workflow-pod-brain/src/main/scala/service/KubernetesClientService.scala

```diff
@@ -2,33 +2,62 @@ package service
 
 import config.ApplicationConf.appConfig
 import io.kubernetes.client.openapi.{ApiClient, Configuration}
-import io.kubernetes.client.openapi.apis.CoreV1Api
-import io.kubernetes.client.openapi.models.{V1Pod, V1PodList}
+import io.kubernetes.client.openapi.apis.{AppsV1Api, CoreV1Api}
+import io.kubernetes.client.openapi.models.{V1Container, V1ContainerPort, V1Deployment, V1DeploymentSpec, V1LabelSelector, V1ObjectMeta, V1Pod, V1PodList, V1PodSpec, V1PodTemplateSpec}
 import io.kubernetes.client.util.Config
 import service.KubernetesClientConfig.kubernetesConfig
 
+import java.util
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
 object KubernetesClientConfig {
 
   val kubernetesConfig = appConfig.kubernetes
-  def createKubernetesClient(): CoreV1Api = {
+  def createKubernetesCoreClient(): CoreV1Api = {
     val client: ApiClient = Config.fromConfig(kubernetesConfig.kubeConfigPath)
     Configuration.setDefaultApiClient(client)
     new CoreV1Api(client)
   }
+  def createKubernetesAppsClient(): AppsV1Api = {
+    val client: ApiClient = Config.fromConfig(kubernetesConfig.kubeConfigPath)
+    Configuration.setDefaultApiClient(client)
+    new AppsV1Api(client)
+  }
 }
 
 class KubernetesClientService(
                                val namespace: String = kubernetesConfig.namespace,
                                val deploymentName: String = kubernetesConfig.workflowPodPoolDeploymentName) {
 
-  private val api: CoreV1Api = KubernetesClientConfig.createKubernetesClient()
+  private val coreApi: CoreV1Api = KubernetesClientConfig.createKubernetesCoreClient()
```

Comment:
Can you add some comments to explain the difference between these two api?

Reply:
Assuming you mean the difference between the CoreV1Api and AppsV1Api, they are collections of different groups of Kubernetes objects and functions to interact with said objects. The CoreV1Api contains objects and resources that are core building blocks of a K8s cluster, such as pods, namespaces, and services. On the other hand, the AppsV1Api contains a set of higher level application-focused resources such as Deployments and StatefulSets.

Reply:
I have added a commit containing comments which should explain the main difference between the two APIs.